Index: exec_articles.php
===================================================================
--- exec_articles.php	(revision 5669)
+++ exec_articles.php	(working copy)
@@ -106,6 +106,8 @@
 			  $message, $texte_petition,
 			  $changer_virtuel, $virtuel);
  
+echo pipeline('ajouter_boite_gauche',array('date'=>'', 'args'=>array('exec'=>'articles','id_article'=>$id_article,'id_rubrique'=>$id_rubrique));
+
 //
 // Affichage de la colonne de droite
 //
