Index: ecrire/inc_barre.php
===================================================================
--- ecrire/inc_barre.php	(revision 5670)
+++ ecrire/inc_barre.php	(working copy)
@@ -15,8 +15,8 @@
 function afficher_barre($champ, $forum=false) {
 	global $barre_typo;
 	tester_variable("barre_typo", "spip");
-	if($barre_typo != '' AND file_exists(_DIR_RESTREINT."inc_barre_$barre_typo.php" )) {
-		include_ecrire("inc_barre_$barre_typo");
+	if ($barre_typo != '' AND $path_barretypo = find_in_path("inc_barre_$barre_typo.php")) {
+		include_ecrire($path_barretypo);
 		$f = "afficher_barre_$barre_typo";
 		if(function_exists($f)) {
 			return $f($champ, $forum);
Index: ecrire/inc_presentation.php
===================================================================
--- ecrire/inc_presentation.php	(revision 5670)
+++ ecrire/inc_presentation.php	(working copy)
@@ -1809,6 +1809,8 @@
 	. 'spip_style_invisible.css" title="invisible" />' . "\n"
 	. '<link rel="stylesheet" href="' . _DIR_IMG_PACK
 	. 'spip_style_visible.css"  title="visible" />' . "\n"
+	
+	. pipeline('ajouter_styles', '')
 
 	// favicon.ico
 	  . '<link rel="shortcut icon" href="' . _DIR_IMG_PACK . "favicon.ico\" />\n";
